### DISCUSSION TITLE
CMSIS DSP license

### DISCUSSION THREAD

**Discussion by mariuszmatusiak** [ARM-software/CMSIS_5#1404] [Q&A] [answered]
Hi,

I would like to ask about the CMSIS license. I am developing a library that I would like to release as a binary (at least for now). A small part of the code is using some CMSIS DSP functions, e.g. arm_fill_*, arm_copy_*, arm_rftt_*, arm_scale_*, arm_conv_partial_*. In the License section of the documentation, I read that CMSIS is available free of charge under the Apache 2.0 license. I would like to make sure though if I am allowed to use the library in such a distribution. Are there any practical conditions?

**Comment by JonatanAntoni** [comment, marked as answer]
Hi @mariuszmatusiak,

Yes, you are allowed to use CMSIS (including CMSIS-DSP) freely under the conditions of Apache 2.0 which include binary distribution. 
Redistributing source code taken from CMSIS must retain the original copyright and license information.

You might consider adding a third party license information section to your projects license file. There you could state the compile-time dependency to CMSIS and its license. I'd consider this absolutely sufficient to follow Apache 2.0 requirements.

Does this answer your question?

Thanks,
Jonatan

**Reply by mariuszmatusiak** [reply]
Yes, @JonatanAntoni, thank you.
